NextcloudTalk/CallViewController.swift

```diff
@@ -0,0 +1,2392 @@
+        }
+    }
+
+    // MARK: - RTCVideoViewDelegate
+
+    func videoView(_ videoView: any RTCVideoRenderer, didChangeVideoSize size: CGSize) {
+        guard let mtlVideoView = videoView as? RTCMTLVideoView else { return }
+
+        DispatchQueue.main.async {
+            for (peerIdentifier, rendererView) in self.videoRenderersDict.filter({ $0.value == mtlVideoView }) {
+                rendererView.frame = CGRect(origin: .zero, size: size)
+
+                if let indexPath = self.indexPath(forPeerIdentifier: peerIdentifier),
+                   let participantCell = self.collectionView.cellForItem(at: indexPath) as? CallParticipantViewCell {
+
+                    participantCell.setRemoteVideoSize(size)
+                }
+            }
+
+            for (_, rendererView) in self.screenRenderersDict.filter({ $0.value == mtlVideoView }) {
+                rendererView.frame = CGRect(origin: .zero, size: size)
+
+                if self.screensharingView.contentView == rendererView {
+                    self.screensharingSize = rendererView.frame.size
+                    self.screensharingView.contentViewSize = rendererView.frame.size
+                    self.screensharingView.resizeContentView()
+                }
+            }
+        }
+    }
+
+    // MARK: - Cell updates
+
+    @nonobjc
+    private func indexPathAndPeer(with predicate: (NCPeerConnection) -> Bool) -> (indexPath: IndexPath, peer: NCPeerConnection)? {
+        for peerIndex in peersInCall.indices {
+            let peer = peersInCall[peerIndex]
+
+            if predicate(peer) {
+                return (IndexPath(row: peerIndex, section: 0), peer)
+            }
+        }
+
+        return nil
+    }
+
+    private func peerConnection(with predicate: (NCPeerConnection) -> Bool) -> NCPeerConnection? {
+        for peer in peersInCall where predicate(peer) {
+            return peer
+        }
+
+        for peer in screenPeersInCall where predicate(peer) {
+            return peer
+        }
+
+        return nil
+    }
+
+    internal func indexPath(forPeerIdentifier peerIdentifier: String) -> IndexPath? {
+        return self.indexPathAndPeer(with: { $0.peerIdentifier == peerIdentifier })?.indexPath
+    }
+
+    internal func indexPath(forPeerId peerId: String) -> IndexPath? {
+        return self.indexPathAndPeer(with: { $0.peerId == peerId })?.indexPath
+    }
+
+    internal func peerConnection(forPeerIdentifier peerIdentifier: String) -> NCPeerConnection? {
+        return self.peerConnection(with: { $0.peerIdentifier == peerIdentifier })
+    }
+
+    internal func peerConnection(forPeerId peerId: String) -> NCPeerConnection? {
+        return self.peerConnection(with: { $0.peerId == peerId && $0.roomType == kRoomTypeVideo })
+    }
+
+    internal func screenPeerConnection(forPeerId peerId: String) -> NCPeerConnection? {
+        return self.peerConnection(with: { $0.peerId == peerId && $0.roomType == kRoomTypeScreen })
+    }
```

Comment:
This is a "breaking" change to the code we had before. Should discuss if it's fine like this.